ggshield/cmd/honeytoken/create.py

```diff
@@ -0,0 +1,97 @@
+import random
+import string
+import sys
+from typing import Any, Dict, Optional
+
+import click
+from pygitguardian import GGClient
+from pygitguardian.models import Detail, HoneytokenResponse
+
+from ggshield.cmd.common_options import add_common_options
+from ggshield.core.client import create_client_from_config
+from ggshield.core.config import Config
+from ggshield.core.errors import UnexpectedError
+from ggshield.core.utils import urljoin
+
+
+def _generate_random_honeytoken_name() -> str:
+    """Generate a honeytoken name based on a random string of eight alphanumeric characters"""
+    letters_and_digits = string.ascii_letters + string.digits
+    random_str = "".join(random.choice(letters_and_digits) for i in range(8))
+    return f"ggshield-{random_str}"
+
+
+def _dict_to_string(data: Dict, space: bool = False) -> str:
+    """Returns a string with 'key=value' for each key-value pair in the dictionary."""
+    space_char = " " if space else ""
+    return "\n".join([f"{k}{space_char}={space_char}{v}" for k, v in data.items()])
+
+
+def _get_honeytoken_dashboard_url(
+    config: Config, honeytoken: HoneytokenResponse
+) -> str:
+    return urljoin(config.workspace_dashboard_url, "honeytokens", str(honeytoken.id))
+
+
+@click.command()
+@click.option(
+    "--name",
+    "name",
+    required=False,
+    type=str,
+    help="give a name to your honeytoken. Default ggshield-xxxxxxxx",
+)
+@click.option(
+    "--type",
+    "type_",
+    required=True,
+    type=click.Choice(("AWS",)),
+)
+@click.option(
+    "--description", "description", required=False, type=str, help="250 characters max"
+)
+@add_common_options()
+@click.pass_context
+def create_cmd(
+    ctx: click.Context,
+    name: Optional[str],
+    type_: str,
+    description: Optional[str],
+    **kwargs: Any,
+) -> int:
+    """
+    Create a honeytoken
+    """
+    # if name is not given, generate a random one
+    if not name:
+        name = _generate_random_honeytoken_name()
+    config: Config = ctx.obj["config"]
+    client: GGClient = create_client_from_config(ctx.obj["config"])
+    response = client.create_honeytoken(name, type_, description)
+    if not isinstance(response, (Detail, HoneytokenResponse)):
+        raise UnexpectedError("Unexpected honeytoken response")
+
+    if isinstance(response, Detail):
+        raise UnexpectedError(response.detail)
+    honeytoken: HoneytokenResponse = response
+
+    token_to_display = {}
+    for k, v in honeytoken.token.items():
+        token_to_display[f"{honeytoken.type_}_{k}".lower()] = v
```

Comment:
We do not use single-letter variables, except in list or dict comprehensions. But you can rewrite it as one :)

```suggestion
    token_to_display = {f"{honeytoken.type_}_{k}".lower(): v for k, v in  honeytoken.token.items()}
```